### PR TITLE
[BCI-2553]: adds starknet rpc support

### PR DESCRIPTION
**Pull request by chris-de-leon-cll**
## Closes

https://smartcontract-it.atlassian.net/browse/BCI-2553

## Description

Updates starknet layer2 sequencer health adapter to use RPCProvider instead of SequencerProvider

## Changes

- Updates starknet.js dependency to latest 5.x version
- Removes sequencer and gateway endpoints in favor of RPC endpoint
- Fixes empty transaction error handling

## Steps to Test

1. Unit tests
2. Ran the adapter locally and ran the request to check Starknet health status

```sh
export adapter=layer2-sequencer-health && yarn test $adapter/test/unit && yarn test $adapter/test/integration
```

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `adapter-secrets` configuration file or update the [soak testing blacklist](/packages/scripts/src/get-changed-adapters/soakTestBlacklist.ts).
- [x] If a new adapter was made, or a new endpoint was added, update the `test-payload.json` file with relevant requests.
	- N/A
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Jira.
- [x] This is related to a maximum of one Jira story or GitHub issue.
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types).
- [x] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

**Comment by augustbleeds** [comment]
Nice~ Did you test that it works agains the real RPC endpoint?

**Comment by augustbleeds** [comment]
> Nice~ Did you test that it works agains the real RPC endpoint?

I got back
```
{
    "jobRunID": "1",
    "result": 0,
    "statusCode": 200,
    "data": {
        "result": 0
    },
    "metricsMeta": {
        "feedId": "{\"data\":{\"network\":\"starkware\"}}"
    }
}
```

So it seems good!

**Comment by chris-de-leon-cll** [comment]
@augustbleeds Yup I tested it locally using one of the RPC endpoints from [their docs](https://docs.starknet.io/documentation/tools/api-services/)

**Review by augustbleeds — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4262,7 +4262,7 @@ __metadata:
     ethers: ^5.4.6
     nock: 13.2.9
     sinon: 12.0.1
-    starknet: 5.8.0
+    starknet: 5.24.3
```
Comment (archseer) [inline yarn.lock:4265]: You should actually use the latest `6.x` beta since it supports newer RPC versions

Comment (archseer) [reply yarn.lock:4265]: Looks like v6 might be out https://github.com/starknet-io/starknet.js/commit/260e77ac0e8670cc2b22eda8067854c7bc3970b5

Comment (chris-de-leon-cll) [reply yarn.lock:4265]: @archseer  sounds good! I've upgraded the package to 6.0.0-beta.15 - just a heads up there were a few build errors that I ran into once I did this

The latest 6.x beta version depends on [this](https://www.npmjs.com/package/abi-wan-kanabi) package, and if the code is compiled via `yarn setup` with the `noUnusedLocals` and `noUnusedParameters` options set to true, then typescript throws some errors (see screenshot below)

For now I've disabled these options in `/packages/sources/layer2-sequencer-health/tsconfig.json` and left a `TODO` to remove them once the compile errors are resolved. However, if this is not ideal, I'm happy to use a different approach to resolve the issue just lmk

Screenshot 2024-02-08 at 2 50 57 PM

**Review by krebernisak — APPROVED** [review]

**Review by alejoberardino — APPROVED** [review]

**Pull request merged**